### PR TITLE
1660 - Link Query Examples to Terms

### PR DESCRIPTION
**Pull request by Vladysl**

### CODE REVIEW

File: odd-platform-ui/src/lib/hooks/api/terms.ts [hunk 1]
```diff
@@ -29,3 +35,45 @@ export function useGetTermLinkedColumns(params: TermApiGetTermLinkedColumnsReque
...
+        exampleId,
+      }),
+    onSuccess: async () => {
+      showSuccessToast({ message: 'Query Example successfully unassigned!' });
+      await queryClient.invalidateQueries({
+        queryKey: ['getQueryExamplesByTermId'],
+      });
+    },
+  });
+}
```
Comment (AndreyNenashev) [inline odd-platform-ui/src/lib/hooks/api/terms.ts:79]: We don't need to use arg for termId at this level. So mutation func with two args is more obvious behaviour. Same for the `useAssignTermQueryExample` hook
```suggestion
export function useUnassignTermQueryExample() {
  const queryClient = useQueryClient();

  return useMutation({
    mutationKey: ['unassignTermQueryExample'],
    mutationFn: async ({
      exampleId,
      termId
    }: TermApiDeleteQueryExampleToTermRelationshipRequest) =>
      termApi.deleteQueryExampleToTermRelationship({
        termId,
        exampleId,
      }),
    onSuccess: async () => {
      showSuccessToast({ message: 'Query Example successfully unassigned!' });
      await queryClient.invalidateQueries({
        queryKey: ['getQueryExamplesByTermId'],
      });
    },
  });
}
```

**Review by AndreyNenashev — CHANGES_REQUESTED** [review]

**Review by AndreyNenashev — APPROVED** [review]

File: odd-platform-ui/src/components/shared/elements/QueryExamples/QueryExamplesListItem.tsx [hunk 2]
```diff
@@ -1,61 +1,87 @@
...
+import QueryExamplesTermUnlinkButton from './QueryExamplesTermUnlinkButton';
 
 interface QueryExampleSearchResultsItemProps {
   queryExample: QueryExample;
-  dataEntityId?: number;
+  isTerm?: boolean;
+  entityId?: number;
 }
 
 const QueryExamplesListItem = ({
```
[the remark this answers is not in the document]
Comment (Vladysl) [reply odd-platform-ui/src/components/shared/elements/QueryExamples/QueryExamplesListItem.tsx:21]: seems like all is fine

File: odd-platform-ui/src/components/Terms/TermDetails/TermQueryExamples/AssignTermQueryExampleForm.tsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  const formId = 'assign-term-query-example-form';
+
+  const { handleSubmit, control, reset, formState } = useForm<FormData>({
+    mode: 'onChange',
+    reValidateMode: 'onChange',
+  });
+
+  const clear = () => {
+    reset();
+  };
```
[the remark this answers is not in the document]
Comment (Vladysl) [reply odd-platform-ui/src/components/Terms/TermDetails/TermQueryExamples/AssignTermQueryExampleForm.tsx]: done

File: odd-platform-ui/src/lib/hooks/api/terms.ts (hunk 1 above)
Comment (Vladysl) [reply odd-platform-ui/src/lib/hooks/api/terms.ts]: done

**Review by AndreyNenashev — APPROVED** [review]

**Pull request merged**